src/checkpoints.cpp

```diff
@@ -30,28 +31,26 @@ namespace Checkpoints {
     }
 
     //Memory only!
-    void AddCheckPoint(const CCheckpointData& data, int64_t height, uint256& hash)
+    void AddCheckPoint(const CCheckpointData& data, int64_t height, uint256 hash)
     {
-        const MapCheckpoints& checkpoints = data.mapCheckpoints;
+        MapCheckpoints& checkpoints = const_cast<MapCheckpoints&>(data.mapCheckpoints);
         checkpoints.insert(std::pair<int64_t,uint256>(height, hash));
     }
 
     //Memory only!
-    bool IsBadpoint(const CBadpointData& data, int64_t height, uint256& hash) 
+    bool IsBadpoint(const CBadpointData& data, int64_t height, uint256 hash) 
     {
         const MapCheckpoints& badpoints = data.mapBadpoints;
+        std::map<int64_t, uint256>::iterator result = const_cast<MapCheckpoints&>(badpoints).find(height);
+        if(result != badpoints.end())
+            return result->second == hash;
+        return false;        
     }
 
     //Memory only!
-    void AddBadPoint(const CBadpointData& data, int64_t height, uint256& hash) 
+    void AddBadPoint(const CBadpointData& data, int64_t height, uint256 hash) 
     {
-        if(data.mapBadpoints) {
-            badpoints.insert(std::pair<int64_t,uint256>(height, hash));
-        } else {
-            data.mapBadpoints = (CCheckpointData){
-                boost::assign::map_list_of(height, hash)
-            };
-        }
+        const_cast<MapCheckpoints&>(data.mapBadpoints).insert(std::pair<int64_t,uint256>(height, hash));
```

Comment:
The CBadpointData in CChainParams is not a pointer (and is never nullptr) and is initialized as a zero size Map.  All we need to do is start inserting.